stock_multi_warehouse_security/models/stock_quant.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2019 Akretion
+# Copyright 2022 Foodles (http://www.foodles.co).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    warehouse_id = fields.Many2one(
+        "stock.warehouse", compute="_compute_warehouse_id", store=True, index=True
+    )
+
+    @api.depends("location_id")
```

Comment:
I guess I would depend on `location_id.warehouse_id` here also.
Also, I am not sure why not to have a related field here, it is possible the comment is obsolete

Reply:
I've moved to a related field then !